packages/relay-runtime/store/RelayPublishQueue.js

```diff
@@ -57,8 +57,15 @@ type PendingUpdater = {|
   +updater: StoreUpdater,
 |};
 
+const _global =
+  typeof global !== 'undefined'
+    ? global
+    : typeof window !== 'undefined'
+    ? window
+    : undefined;
```

Comment:
i feel like this is used in quite a number of other places, and very idiomatic. 

I suspect, just `global` as it is is fine, and handle this in your own eco with webpack overrides or babel plugins.

Reply:
Just checking `global` without testing for existence breaks in non-Node.js environments. In this case Relay is broken on Vercel's new runtime that doesn't use Node.js and has no access to `global`. Using `globalThis` would fix the issue as well but I'm not sure that would have been a simple direct swap. This assignment to `_global` is copied directly from another place in the Relay codebase that also looks for global variables.

Regarding handling this in Webpack or Babel plugins: that would require additional processing to `node_modules` which can be tricky to set up in metaframeworks. In the case of `relay-nextjs` I would also have to add instructions on how to do that in the onboarding guide, which adds another step to an already lengthy process.

Reply:
My 2 cents here: many environments such as Web Worker don't have `window` or `global`. And there're also non-Node.js and non-browser environments like RN and Deno. I think using the `globalThis` standard will make it a lot easier.

One potential problem is, `globalThis` was introduced since Node.js 12 (same for browsers). But it should be good if we check it with `typeof`.

Reply:
@shuding I think replacing all usages of `global` in the Relay codebase might be out of scope of this PR 😅 I agree that it's probably better but there's a good chance that a flat-out replacement may not work in FB's internal system.